### PR TITLE
Fix 49974 attachment infinite loading

### PR DESCRIPTION
**Pull request by Kalydosos**
### Fixed Issues

$ https://github.com/Expensify/App/issues/49974
PROPOSAL: https://github.com/Expensify/App/issues/49974#issuecomment-2389704800

### Tests

1. Launch the app. 
2. Open a chat
3. Paste the text Alt text
4. Send the message
5. Click on the image
6. Verify that an icon and the text "No Attachment Found" are displayed

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/5725c05c-3b5b-4ebb-ae35-424fbb3aef34
</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/7425877e-2d24-469c-8ea6-2259c3124a2a
</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/98906f54-cac9-4800-b59d-53819a1a9965
</details>

<details>
<summary>iOS: mWeb Safari</summary>

ios_mweb_safari
</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/a0138782-b3c7-419f-a0be-7c3ac01a5dfc
</details>

<details>
<summary>MacOS: Desktop</summary>

macos_desktop
</details>

### CODE REVIEW

**Comment by Kalydosos** [comment]
@hoangzinh Hi, i'm pinging you about the translation process. We need to translate the wording "Attachment not found" in spanish (alone ?). Thx

**Comment by Kalydosos** [comment]
@Expensify/design will be needed for review

**Comment by Kalydosos** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/components/Attachments/AttachmentView/index.tsx
Comment (Kalydosos) [reply src/components/Attachments/AttachmentView/index.tsx:236]: OK, you mean to have both the icon and the text in the same View ? Normally the fallback image should always be found as it is hardcoded and not a "parameter".  Dont you think ?

Comment (Kalydosos) [reply src/components/Attachments/AttachmentView/index.tsx:236]: @hoangzinh we deal only with AttachmentView and our fallbackIcon is hardcoded as 
https://github.com/Kalydosos/App/blob/59ca13c87ddd8a552a8bbe69191dedb24315bbed/src/components/Attachments/AttachmentCarousel/CarouselItem.tsx#L87
we will neverbe using  fallbackIcon = Expensicons.FallbackAvatar

I would love to put falbackText eventually the PR is overdue. already and i will not have the time to restest everything

Comment (Kalydosos) [reply src/components/Attachments/AttachmentView/index.tsx:236]: @hoangzinh once we agreed on this part, i suggess i set the PR ready for review and delete this thread of conversation to keep the reviewers focused on the solution. What do you think ?

**Comment by Kalydosos** [comment]
@MelvinBot recheck

**Comment by Kalydosos** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by shawnborton** [comment]
For this screenshot here:
CleanShot 2024-10-18 at 16 13 25@2x

I think the icon above the headline text should use our icon color, right @Expensify/design? It looks a little light, so just want to double check. Thanks!

File: src/components/Attachments/AttachmentView/index.tsx
Comment (shawnborton) [inline src/components/Attachments/AttachmentView/index.tsx:231]: Consider making this theme.icons instead

**Comment by dannymcclain** [comment]
>I think the icon above the headline text should use our icon color, right @Expensify/design?

Yes that's correct!

File: src/components/Attachments/AttachmentView/index.tsx
Comment (Kalydosos) [reply src/components/Attachments/AttachmentView/index.tsx:231]: Ok, thx for the feedback

I tried theme.icon but it"s still light as you can see in the pic below
Capture d’écran de 2024-10-18 20-50-47

but a theme that seems to match better the "Attachment not found" text in both light and dark modes is theme.iconHovered as you can see in the pic below

Capture d’écran de 2024-10-18 20-57-58

What do you think ? also @dannymcclain

Comment (dannymcclain) [reply src/components/Attachments/AttachmentView/index.tsx:231]: The first pic (`theme.icon`) looks correct!
image

Comment (Kalydosos) [reply src/components/Attachments/AttachmentView/index.tsx:231]: Ok @dannymcclain thx for the feedback, @shawnborton wha's your decision ?

Comment (shawnborton) [reply src/components/Attachments/AttachmentView/index.tsx]: Yup, Danny is correct!

Comment (Kalydosos) [reply src/components/Attachments/AttachmentView/index.tsx]: @shawnborton yes, i have updated the code already. thx for the feedback :+1:

Comment (shawnborton) [reply src/components/Attachments/AttachmentView/index.tsx]: Thanks!

**Comment by Kalydosos** [comment]
@hoangzinh @chiragsalian I have seen some ESLint issues on the runs https://github.com/Expensify/App/actions/runs/11429650492 y i help through updates to the code ? I can eventually fix the first issue by replacing with import {AttachmentNotFound} from '@components/Icon/Expensicons'  (but we could not list all icons in the long run) and i could also add extra ";" to the translation files but for the other issues i really dont see how they are relevant as the same patterns of synatx is everywhere in the file. Let me know if anything i can do ? Thx

**Comment by Kalydosos** [comment]
> For example, it means we need to move Line 5 to Line 7 Screenshot 2024-10-22 at 17 22 04

ok, i got it; i'll update the code, thx :+1:

**Comment by Kalydosos** [comment]
> Bug 1: App shows "Attachment not found" error instead of an "Offline" message when view images in Offline mode.
> 
>  Screen.Recording.2024-10-21.at.17.57.19.mov 
> Click here to view before changes

@hoangzinh  ok, i'll fix that, thx for the feedback :+1: 

Also, in relation to the choices in https://github.com/Expensify/App/issues/49974#issuecomment-2411727068, it turns out that we have no way to know if there is a problem with the image address once offline, so we cannot say "Attachment not found" when offline and clicking on an image loaded with error. The reason is the same why we also say "Offline" despite clicking on a perfectly loaded image but being offline. What do you think ? I am basically saying that once offline either the image was previously loaded with or without error we have no way to know that when offline (we cannot assess the validity of the url at that moment in fact) so we can only notice the user of being offline. 

I will sum up then the different scenarii ;

1. If the image was loaded with an error and when clicked upon : the network is online => "Attachment not found" view
2. If the image was loaded with an error and when clicked upon : the network is offline => -"Attachment not found" view- Offline view
3. If the image was loaded without error and when clicked upon : the network is online => image displayed
4. If the image was loaded without error and when clicked upon : the network is offline => Offline view
5. if the image is loaded when offline and clicked upon when offline  => image displayed 

What do you think ?

**Comment by Kalydosos** [comment]
> @Kalydosos can you add test steps for "Offline tests" and "QA tests"? Thank you

@hoangzinh Hi, i have added some offline tests steps. I will modify them if necessary according to what you think of my point on offline status. I don't see any relevant QA test to add. Do you think of any ? Thx

**Comment by Kalydosos** [comment]
> Hi @Kalydosos do you still have a question [here](https://github.com/Expensify/App/pull/50511#issuecomment-2430512561) or you had an answer for it?
> 
> > I don't see any relevant QA test to add
> 
> QA test is used for QA team to follow to test this PR. I think in this case, it can be same as "Test steps"

@hoangzinh no that was the point i need that we agreed on. Is it ok for you ?

**Comment by Kalydosos** [comment]
> @Kalydosos can you help to merge the latest main branch into this PR? There is a node upgrade version merged, and we need to keep our PR update to ensure everything work well #50928

@hoangzinh ok i will do that

File: src/libs/fileDownload/FileUtils.ts
Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: > @Kalydosos can you clarify a bit why we need this change here? Thanks

@hoangzinh already loaded images before going offline have an adress that starts with "/chat-attachments'" and with the previously used function  
https://github.com/Kalydosos/App/blob/92948a59b9a9b49ba6c483bf29d74dd194b18775/src/libs/fileDownload/FileUtils.ts#L297-L302
they were seen as "local" files and when offline, the offline view was not displayed when clicked on the images. The new function was used here  https://github.com/Kalydosos/App/blob/92948a59b9a9b49ba6c483bf29d74dd194b18775/src/components/ImageView/index.tsx#L238-L239  . I could not reused https://github.com/Kalydosos/App/blob/92948a59b9a9b49ba6c483bf29d74dd194b18775/src/libs/fileDownload/FileUtils.ts#L297-L302  in the new function, it was safer to copy and adapt it

**Comment by Kalydosos** [comment]
> > @Kalydosos can you help to merge the latest main branch into this PR? There is a node upgrade version merged, and we need to keep our PR update to ensure everything work well #50928
> 
> @hoangzinh ok i will do that

@hoangzinh i have done the merge and test it. The images loaded with error are not clickable anymore. I guess it's a workaround. Should i remove that workaround so the images become clickable and we can provide the users with a more clear and complete context ?

Capture d’écran de 2024-10-24 19-43-29

**Comment by Kalydosos** [comment]
@hoangzinh i have made the images clickable and push the changes. The PR is fully complete normally (if 1 or 2 Eslint recommandations dont rermain).

**Comment by Kalydosos** [comment]
> > The images loaded with error are not clickable anymore
> 
> It has been intended behavior since this PR #50918. We can leave it as it is. Can you revert your change in [3320b50](https://github.com/Expensify/App/commit/3320b50b34c86fd17a45b04282a10001a4dccb08)

@hoangzinh they expect our PR to fix it https://github.com/Expensify/App/pull/50918#issuecomment-2423962173, making it disabled was a workaround, i think we should make it clickable to provide more context to the user, that's the whole point of this PR. They are expecting us to https://github.com/Expensify/App/pull/50918#issuecomment-2423962173. What do you think ?


https://github.com/user-attachments/assets/32300f6b-0b8f-4082-a675-9fc82b41387d


https://github.com/user-attachments/assets/1373fd83-1bb9-4b1d-a3b2-d62fbdbf3ebb

File: src/components/Attachments/AttachmentView/index.tsx
Comment (Kalydosos) [reply src/components/Attachments/AttachmentView/index.tsx:277]: i dont really see how is that relevant, but i will do it

Comment (Kalydosos) [reply src/components/Attachments/AttachmentView/index.tsx]: done

File: src/libs/fileDownload/FileUtils.ts
Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: maybe the behaviour changes at sometime but i could not reproduce the "before changes" of your video  https://github.com/Expensify/App/pull/50511#issuecomment-2426341037 unless i apply that modification but i was played by my browser cache in the beginning, but once disabled the cache i could not reproduce the "before changes" unless  i made that modification. On whch version did you made the video ?

**Comment by Kalydosos** [comment]
> They're expecting us to do when open the attachment view (carousel) and navigate to the attachment not found :🤔

we cannot open the Attachment view if we cannot click. Am i missing somehing ?


https://github.com/user-attachments/assets/842e996e-1b9e-43dc-929d-b46d3df18067

**Comment by Kalydosos** [comment]
> There is a recording to demonstrate the bug that they're expecting us to fix in this PR [#50918 (comment)](https://github.com/Expensify/App/pull/50918#issuecomment-2423654000)

it seems no diffirent of what we have done. But i will take a closer look. Are we going to deliver this PR someday :smiley: ? 

https://github.com/user-attachments/assets/b734df91-a4e3-45c2-9881-2050df7c5f30

File: src/libs/fileDownload/FileUtils.ts
Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: > I couldn't find previous version

i mean a version or a tag of the app where i can reproduce that : clicking on a well loaded attachment before going offline will display "offliex view" once clicked and offline

**Comment by Kalydosos** [comment]
> Yep, but we can click on the next attachment which is loadable.
> 
> Step to reproduce:
> 
> 1. Go to any chat report
> 2. Send the text Alt text
> 3. Send another real image
> 4. Click on the real image to open attachment view
> 5. Click on the "Previous" arrow button on the left to navigate to the image that we sent in step 2
> 
>  Screen.Recording.2024-10-25.at.21.44.14.mov

ok i will look at it

**Comment by Kalydosos** [comment]
> > Are we going to deliver this PR someday
> 
> Yep, definitely @Kalydosos. We're nearly done. Just 2 concerns above then we're good

ok i will look through those 2 points

File: src/libs/fileDownload/FileUtils.ts
Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: > > already loaded images before going offline have an adress that starts with "/chat-attachments'"
> 
> I think it's not correct @Kalydosos. If the attachment comes from chat report, it will have start with `/chat-attachments`. For example: `/chat-attachments/9208320899643172368/w_cf3e867bf4b45d8dcc5d8168166247b2b44e8a8f.png`
> 
> Therefore, the function name `isOfflineFile` does not really make sense. I think we can either update the existing "isLocalFile" function or add a new function to check `reportAttachmentFile` and use combine it with "isLocalFile". What do you think?

the function was named like that beacuse it only returns true for files loaded when offline in fact. To match this scenario

if the image is loaded when offline and clicked upon when offline => image displayed

>  we can either update the existing "isLocalFile" function or add a new function to check `reportAttachmentFile` and use combine it with "isLocalFile".

i will see for this once i grasp the changes you are talking about in the comments below.

Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: > I see. Do you think it's an existing bug in the main branch?

@hoangzinh i have done the test on the main branch and it turns out that the situation below is the same once you disable the browser cache (btw your cache was enable during your test https://github.com/Expensify/App/pull/50511#issuecomment-2426341037 that's why the problem didnt occur)

> already loaded images before going offline have an adress that starts with "/chat-attachments'" and with the previously used function
https://github.com/Kalydosos/App/blob/92948a59b9a9b49ba6c483bf29d74dd194b18775/src/libs/fileDownload/FileUtils.ts#L297-L302
they were seen as "local" files and when offline, the offline view was not displayed when clicked on the images.

our PR will fix that issue by the same time

Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: > I think it's not correct @Kalydosos. If the attachment comes from chat report, it will have start with `/chat-attachments`. For example: `/chat-attachments/9208320899643172368/w_cf3e867bf4b45d8dcc5d8168166247b2b44e8a8f.png`

yes i speaking of report attachment in the context when the user add them in a chat thread as specify by the ticket https://github.com/Expensify/App/issues/49974

Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: > Therefore, the function name isOfflineFile does not really make sense.

i propose to name it isFileLocalToUserDevice because that the only context when it should return true.

> we can either update the existing "isLocalFile" function or add a new function to check reportAttachmentFile and use combine it with "isLocalFile"

the function isLocalFile serves it purpose, it indicates whether a file is local generally speaking base on the structure of its uri, and that function is used by many other components, so i think we should change it. For adding a new function, we only have the uri and its a simple  basic check on its structure. I advise we keep the "isOfflineFile" and hjust rename it isFileLocalToUserDevice

Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: > const isNotAttachmentFile = !url?.startsWith('/chat-attachments');

@hoangzinh  typeof url can be number sometimes, so this part above could bug i think

so maybe add a check on that part ? otherwise it's a good idea to make the change inside the file and i propose we rename the variable const isNotRemoteAttachmentFile (because it is an attachment file after all but local to the user device not on the server yet because the user is offline)

but it's a good idea :+1:

**Comment by Kalydosos** [comment]
> > it seems no diffirent of what we have done from our point of focus #49974
> 
> Yes, it's correct. That is the point we're focusing to fix @Kalydosos. Just revert change here [3320b50](https://github.com/Expensify/App/commit/3320b50b34c86fd17a45b04282a10001a4dccb08), then we focus on fixing this issue [#50511 (comment)](https://github.com/Expensify/App/pull/50511#issuecomment-2438022266)

About reverting the change here, meaning disable the click on the error image from the chat thread, i think on a UX standpoint it's always a better thing to allow the user to click to have more context and information than to prevent him/her to do so. So i want to recommend that we talk this over with UX/UI team and whoever else can make the call for us and decide if we should/must allow the click or not. What about starting to talk with the team of PR #50918 about the idea just to have some feedback ?  If you think for some reason that is not a good idea, i will totally follow your position on this point and we will revert the changes, no problem at all for that

File: src/libs/fileDownload/FileUtils.ts
Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: @hoangzinh i will also try to come up with a proposition along that idea :+1:

**Comment by dannymcclain** [comment]
Sorry catching up here. I could go either way when it comes to disabling the click or letting the user open the bad image into the "Attachment not found" view.

I personally don't see a huge amount of value in the "Attachment not found" view, but I also don't think it's terrible either.

@Expensify/design do you all have any strong feelings?

**Comment by Kalydosos** [comment]
> Sorry catching up here. I could go either way when it comes to disabling the click or letting the user open the bad image into the "Attachment not found" view.
> 
> I personally don't see a huge amount of value in the "Attachment not found" view, but I also don't think it's terrible either.
> 
> @Expensify/design do you all have any strong feelings?

@dannymcclain thx for the feedback. It is not an easy choice indeed. 

@hoangzinh i forgot to ask : if we disable the click, should we change somehow the tests steps of the ticket https://github.com/Expensify/App/issues/49974 (which was mentionning explicitly the click action). Should i add the steps you suggested here https://github.com/Expensify/App/pull/50511#issuecomment-2438022266 to the tests steps or we just stick to the steps in the ticket ? I'm more for sticking to the ticket steps as the extra steps are just a consequence, imho

File: src/libs/fileDownload/FileUtils.ts
Comment (Kalydosos) [reply src/libs/fileDownload/FileUtils.ts:293]: @hoangzinh  i propose this by taking inspiration from your advice


> // isLocalToUserDeviceFile means the file is located on the user device,
    // not loaded on the server yet (the user is offline when loading this file in fact)
    let isLocalToUserDeviceFile = FileUtils.isLocalFile(url);
    if (isLocalToUserDeviceFile && (typeof url === "string") && url.startsWith("/chat-attachments")){
        isLocalToUserDeviceFile = false ;
    }

i included some comments to make the variable meaning clearer

I tested and it works well. Any thought ?

**Comment by Kalydosos** [comment]
> @Kalydosos If we disable the click, I'm afraid that we will fail at step "Click on the image". If we don't disable the click, we can follow steps in the ticket.

yes indeed, i was also fearing it will impact our tests somehow. Ok, thx :+1: 

i pushed the  change about the isLocalFile function, i followed your advice, it's way better and raise less concerns than having a new function quite similar to isLocalFile, thx :+1:

File: src/components/ImageView/index.tsx
Comment (Kalydosos) [reply src/components/ImageView/index.tsx:205]: honestly, i think a safer and clearer notation like i proposed is better,as some expressions can become quite fuzzy when code must evolve and easily let slip in some bug, i know what i'm talking about believe me

File: src/components/ImageView/index.tsx
Comment (Kalydosos) [reply src/components/ImageView/index.tsx]: ok, it's done, thx :+1:

File: src/components/ImageView/index.tsx
Comment (Kalydosos) [reply src/components/ImageView/index.tsx]: @hoangzinh it's done :+1:

**Comment by Kalydosos** [comment]
> @Kalydosos Looks good. It appears that we're just waiting for @Expensify/design on this comment [#50511 (comment)](https://github.com/Expensify/App/pull/50511#issuecomment-2442357416)

@hoangzinh as Danny said it is good either way. And as our ticket requires it, we should made the images clickable. Let's follow rule#1 (get shit done)

**Comment by Kalydosos** [comment]
Hi @MariaHCD, to add something to @hoangzinh statement, the other PR was finally complementary of ours and was mainly about changing the gray icon for failed previews https://github.com/Expensify/App/pull/50918 and couldnt handle all aspects of the question as they mentionned https://github.com/Expensify/App/pull/50918#issuecomment-2423962173 but in the process they disable the click on the failed preview which our PR ask us explicitly to fix including the rendering in the carousel

the end result of us renabling the click will be the following


https://github.com/user-attachments/assets/32300f6b-0b8f-4082-a675-9fc82b41387d


https://github.com/user-attachments/assets/1373fd83-1bb9-4b1d-a3b2-d62fbdbf3ebb

**Comment by MariaHCD** [comment]
I'm ooo today - I will catch here tomorrow!

**Comment by MariaHCD** [comment]
Thanks for the context here. If the design team is fine either way, I think re-enabling the click makes sense. It would make it very clear to the user that something is wrong with the attachment.

**Review by hoangzinh — COMMENTED** [review]
@Kalydosos Overal looks good. Can you fix eslint errors and merge the latest main to this PR? Thank you

File: src/components/ImageView/index.tsx
Comment (Kalydosos) [reply src/components/ImageView/index.tsx:204]: @hoangzinh i have done the merge and eslint modifications, thx :+1:

File: src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx
Comment (Kalydosos) [reply src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx:70]: @hoangzinh  no we cant, it's used in the code, here per example 
https://github.com/Expensify/App/blob/f2e295fd174c87f7a516800c22028b637746b22c/src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx#L83
We just dont access the variable hasLoadFailed but it used by useState

File: src/components/ImageView/index.tsx
Comment (Kalydosos) [reply src/components/ImageView/index.tsx]: Doùe and commited, thx :+1:

File: src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx
Comment (Kalydosos) [reply src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.tsx]: @hoangzinh ok i checked, it's done, thx :+1:

**Review by hoangzinh — APPROVED** [review]
LGTM. @Kalydosos congrats on your first PR

**Review by MariaHCD — APPROVED** [review]
LGTM! We can get this merged once we have a final review from design

**Review by MariaHCD — DISMISSED** [review]

**Comment by MariaHCD** [comment]
Ooops, @Kalydosos two issues:

1. There is an unverified commit `fff2058b246e1f45ae4a0f952a48a9a35175a4e2`:

Screenshot 2024-11-06 at 4 40 34 PM

2. There is an ESLint error:

Screenshot 2024-11-06 at 4 40 46 PM

**Comment by Kalydosos** [comment]
@MariaHCD issue 2 is fixed, i have commited the change. I'm working on fixing issue 1. Thx for the feedback, :+1:

**Comment by dannymcclain** [comment]
Not sure exactly where the latest screenshots/videos are—but I think the styles are all looking good to me. If [these two videos](https://github.com/Expensify/App/pull/50511#issuecomment-2448665860) are the latest, I think it's looking good. 👍

**Comment by Kalydosos** [comment]
@dannymcclain yes these videos are the lastest, thx for the feedback :+1:

**Comment by Kalydosos** [comment]
> @MariaHCD issue 2 is fixed, i have commited the change. I'm working on fixing issue 1. Thx for the feedback, 👍

@MariaHCD i will resign all the commits, which will sign the first one in question but the commits ids will change. I hope this will not break any of the automation of the process

**Comment by Kalydosos** [comment]
> > @MariaHCD issue 2 is fixed, i have commited the change. I'm working on fixing issue 1. Thx for the feedback, 👍
> 
> @MariaHCD i will resign all the commits, which will sign the first one in question but the commits ids will change. I hope this will not break any of the automation of the process as i will have to force push the branch. Any thoughts on that ?

@hoangzinh  any advice also ? I'm about the rebase and force push the branch

**Comment by MariaHCD** [comment]
> It seems PR can't be allowed to merge if there is an unverified commit

Yes, we can't merge a PR that has an unverified commit.

> Or close this PR and open another PR (mention to current PR if anyone wants to back read discussions)

This seems like the best option because force pushing might change the review history. Let's go with this option 👍🏼

**Comment by Kalydosos** [comment]
@MariaHCD @hoangzinh ok, i will set it up right away. thx for your advices :+1:

**Comment by Kalydosos** [comment]
> @Kalydosos we can close this PR

@hoangzinh it's done, thx :+1:

**Comment by Kalydosos** [comment]
> LGTM. @Kalydosos congrats on your first PR

thank you @hoangzinh for your help

**Pull request closed without merging**
